src/NServiceBus.Core/Installation/InstallationComponent.cs

```diff
@@ -0,0 +1,81 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Installation;
+    using Settings;
+    using Transport;
+
+    class InstallationComponent
+    {
+        public InstallationComponent(ReadOnlySettings settings)
+        {
+            this.settings = settings;
+        }
+
+        public void Initialize(List<Type> concreteTypes, ContainerComponent container, ReceiveComponent receiver)
```

Comment:
what about moving all of this code the the ctor and remove the Initialize method?

Reply:
I'm against doing any logic in the ctor (https://stackoverflow.com/questions/24038907/logic-in-constructor sums it up) => don't do anything in there that can throw

All component have a .Initialize method so let's discuss and make any needed change across the board?